### ISSUE TITLE
Move translations to Weblate

### ISSUE THREAD

**Issue by phw** [phw/peek#36]
I have requested for Peek to be hosted on https://hosted.weblate.org, this will make the translation process much smoother.

**Comment by mohamedsakhri** [comment]
I didn't know about weblate. Does it have a significant advantage compared to transifex?

**Comment by phw** [comment]
For me the most important advantage is that it fully integrates with Git. That means any changes to the pot file will automatically be shown on Weblate, and all translations will automatically be committed and pushed back to git. This also sets the translator as the committer, which is also very nice to track contributions.

I also find it easier to navigate and the quality checks are useful. The translation editor is maybe not as modern looking as Transifex', but also good to use.

Let's see, my request to be hosted there still needs to be answered, and that can take a while :D

**Comment by mohamedsakhri** [comment]
Have you took a look at [txgh](https://github.com/transifex/txgh). It's mentioned also in the official website: [Github – Transifex Integration](https://www.transifex.com/blog/2013/github-transifex-integration/).

**Comment by phw** [comment]
Thanks for the tip, looks great and easy enough to set up :+1: Will give it a try before switching to Weblate.

**Comment by phw** [comment]
I have set up txgh and tested it a bit in both directions. looks good so far. If this turns out to be a viable solution I will keep it. Only downside I see so far is that it probably does not support translations changed in source (git) to be synced back to transifex, only pot file changes.

**Comment by mohamedsakhri** [comment]
> Only downside I see so far is that it probably does not support translations changed in source (git) to be synced back to transifex, only pot file changes.
What do you mean exactly? Could you provide a user case as an example?

**Comment by phw** [comment]
What I mean is that txgh currently supports the following two use cases:

1. Translations are completed on Transifex => the translation will be pushed to GitHub
2. A translation source file (the .pot file) is changed in Git (e.g. because new strings where added to the app) => the source file will be pushed to Transifex so new or changed strings are available for translation.

What it does not support is the translations themselves being changed in Git (e.g. by a pull request). This will not trigger the translations gettings synced to Transifex. But that's no too bad, as I prefer getting the translations via Transifex anyway and if necessary I can push myself.

**Comment by mohamedsakhri** [comment]
According to the documentation [[How it works](https://github.com/transifex/txgh)] this use case isn't supported. I think it's the right behaviour and it shouldn't be supported because the services should stay separated as they are defined for. Transifex to translate and git(hub) to handle files versions.

**Comment by phw** [comment]
TL;DR: Go to https://hosted.weblate.org/projects/peek/ to translate Peek, stop using Transifex ;)

Long version:
Transifex has worked ok as a first solution, but it is lacking regarding integration with Git and Github. I had tried an intermediate solution using txgh, but the result was not satisfactory. @nijel from Weblate was kind enough and enabled the Peek project on [Weblate](https://weblate.org/), which gives us a couple of benefits:

- Translations are automatically pushed into Git whenever new translations have been made
- Changes to the translations in the Git repository are immediately shown in Weblate, as are changes to the POT file
- Preview images, showing the actual usage of a string to the translator
- Comprehensive quality checks
- Prioritization of strings
- Translations will be credited to the actual translator in Git
- It is open source, actively developed and the developer is a great guy :)

If you want to continue translating Peek, please do so from now on on https://hosted.weblate.org/projects/peek/ . I know it means yet another account to create, but IMHO it is well worth it.

I have set the Transifex project to private now. Existing translators still have access, and I will monitor the Transifex project for changes for a while, but in a few weeks I will completely shut it down.

If you have any questions or comments, please comment here or write me directly on ph.wolfer@gmail.com.

Thanks again to all who have translated so far,
Philipp

**Comment by phw** [comment]
And we already have a first result, a completed Polish translation :)
https://github.com/phw/peek/commit/85917bb500fa3755c7e11b70fc491cfaf1fba827 shows how a Weblate commit looks like.

**Comment by dawidd6** [comment]
Very nice @phw

**Comment by mohamedsakhri** [comment]
I've given Weblate a try. It has really good features to understand the context of each string, like source information (sometimes with screenshot) and the translation in other languages the translator can understand.
The interface is IMHO not so user friendly like Transifex. It must take some time to get used to it.

@phw Out of topic: Did you to display the issues ordered by last issue first rather than last answer first or is it the default behaviour in Github? I've been active in some projects and they have "last answer first".

**Comment by nijel** [comment]
@mohamedsakhri We indeed lack some UX expert at Weblate, any suggestions how to improve user experience are welcome :-).

**Comment by mohamedsakhri** [comment]
@nijel I'm no UX expert neither and I've only used Weblate for a short time till now. 
For me I have the impressions that there is too much items/entries in the interfaces. It is nice to have many features but than can be disturbing and prevent the user from doing the main work he wants to do: translating.
So my suggestion would be to define a few main goals/features and make them accessible to the user easily in a user friendly interface, simple and clean.
In the screenshot below you can see how "messy" the interface can get. In this case I think "Translations" is the most important part of the page and it should be in focus.

**Comment by nijel** [comment]
Good point, however let's avoid polluting this issue tracker even more and move discussion to Weblate issue tracker: https://github.com/nijel/weblate/issues/1283
